src/spells.cpp

```diff
@@ -623,6 +623,12 @@ ASPELL(spell_dimension_door) {
         char_printf(ch, "You failed.\n");
         return CAST_RESULT_CHARGE;
     }
+    if (victim->in_room == -1) {
```

Comment:
Approving, but I cringe at this.  We should make a function where the type (DD/HG/MW/etc) is passed in.  Duplicate code is bad.  (I know you didn't write it, just saying.)